modules/qt/nixos.nix

```diff
@@ -0,0 +1,57 @@
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}:
+
+let
+
+  recommendedStyle = {
+    gnome = if config.stylix.polarity == "dark" then "adwaita-dark" else "adwaita";
+    kde = "breeze";
+    qtct = "kvantum";
+  };
+
+in
+{
+  options.stylix.targets.qt = {
+    enable = config.lib.stylix.mkEnableTarget "QT" pkgs.stdenv.hostPlatform.isLinux;
+    platform = lib.mkOption {
+      description = ''
+        Platform for QT.
+
+        Defaults to the standard platform of the configured DE.
+
+        Fallback to qtct.
+      '';
+      type = lib.types.str;
+      default = "qtct";
+    };
+  };
+
+  config = lib.mkIf (config.stylix.enable && config.stylix.targets.qt.enable) {
+
+    stylix.targets.qt.platform =
+      if config.services.xserver.desktopManager.gnome.enable then
+        lib.mkDefault "gnome"
+      else if config.services.xserver.desktopManager.plasma5.enable then
+        lib.mkDefault "kde"
+      else if config.services.xserver.desktopManager.lxqt.enable then
+        lib.mkDefault "lxqt"
+      else
+        lib.mkDefault "qt5ct";
```

Comment:
> > I intend to only enable Kvatum if none of the following are enabled:
> >
> > * KDE
> > * GNOME
> > * LXQt
> >
> > Which means I want to make this PR into something that is obsolete for anyone using the above. Just give me a thumbs up if you agree.
>
> Considering the blockers of this PR, it seems good to guard against these known these problems.
>
> -- [NAHO, "kde: replace `kdeglobals` with `Kvantum` theme"](https://github.com/danth/stylix/pull/142#issuecomment-2599687491)

The Home Manager module does not seem to check this. Could you elaborate on this?

Reply:
These desktop environments are usually enabled through NixOS and would as such be invisible to Home-Manager under normal circumstances.
So the strategy was to either inherit the setting from NixOS or fallback to qtct.

Reply:
> So the strategy was to either inherit the setting from NixOS or fallback to qtct.

This would not work when the DEs are not setup with a NixOS configuration that also uses Stylix.

> These desktop environments are usually enabled through NixOS and would as such be invisible to Home-Manager under normal circumstances.

Someone installing using a standalone Home Manager setup with Stylix would not receive these checks.

Also, these checks assume that only of these DEs are enabled, or at least that the prioritization order is: GNOME > KDE > LXQt.

Reply:
Yes, I am aware of it all. Hence, I chose this solution.
The only takeaway for me to improve this so far is:
- give warnings when multiple DE's are enabled (this is problematic for many reasons not just styling as not all settings can be made session dependent)